executor/new_analyze.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package executor
+
+import (
+	"github.com/juju/errors"
+	"github.com/pingcap/tidb/context"
+	"github.com/pingcap/tidb/distsql"
+	"github.com/pingcap/tidb/expression"
+	"github.com/pingcap/tidb/model"
+	"github.com/pingcap/tidb/statistics"
+	"github.com/pingcap/tidb/util/types"
+	"github.com/pingcap/tipb/go-tipb"
+)
+
+var _ Executor = &AnalyzeIndexExec{}
+
+func (e *AnalyzeExec) handleAnalyzePushDown(task *analyzeTask) statistics.AnalyzeResult {
+	if task.taskType != idxTask {
+		return statistics.AnalyzeResult{}
+	}
+	if err := task.src.Open(); err != nil {
+		return statistics.AnalyzeResult{Err: err}
+	}
+	_, err := task.src.Next()
```

Comment:
We can call a function like `getHist()` to get histogram and err directly, which is simpler.